src/util/serialize_exception.cpp

```diff
@@ -47,92 +47,93 @@ namespace hpx { namespace serialization
         std::string throw_state_;
         std::string throw_auxinfo_;
 
-        // retrieve information related to boost::exception
-        try {
-            std::rethrow_exception(ep);
-        }
-        catch (boost::exception const& e) {
-            char const* const* func =
-                boost::get_error_info<boost::throw_function>(e);
+        // retrieve information related to exception_info
+        if (exception_info const* xi = get_exception_info(ep)) {
```

Comment:
Is that a possibility? If I understand the original code correctly, it would simply let the exception propagate in that situation. I can replicate the try/rethrow/catch to maintain the behavior in that case.